File: backend/DvbLiveBackend/ApiStructure/Output/VehiclePositionTripStop.cs
using System;

namespace DerMistkaefer.DvbLive.Backend.ApiStructure.Output
{
    /// <summary>
    /// Api Output for an Trip Stop of the Vehicle Position Output
    /// </summary>
    public class VehiclePositionTripStop
    {
        /// <summary>
        /// Id of the Trip Stop.
        /// </summary>
        public string IdStopPoint { get; set; } = "";
        
        /// <summary>
        /// Arrival Time on this Stop of an Trip. As Unix Timestamp.
        /// </summary>
        public int? ArrivalTime { get; set; }
        
        /// <summary>
        /// Departure Tie on this Stop of an Trip. As Unix Timestamp.
        /// </summary>
        public int? DepartureTime { get; set; }
    }
}

File: frontend/src/services/backend/output/VehiclePositionTripStop.ts
/**
 * Api Output for an Trip Stop of the Vehicle Position Output
 */
export interface VehiclePositionTripStop {
    /**
     * Unique Id for this Stop Point
     */
    idStopPoint: string;

    /**
     * Arrival Time on this Stop of an Trip.
     */
    arrivalTime: number | null;

    /**
     * Departure Tie on this Stop of an Trip.
     */
    departureTime: number | null;
}
